core/arxiv/submission/services/classic/tests/test_get_submission.py

```diff
@@ -157,3 +157,91 @@ def test_get_submission_with_hold_and_reclass(self):
+            SetAbstract(creator=user, abstract='Indeed' * 20),
+            SetAuthors(creator=user, authors=[
+                Author(order=0, forename='Arthur', surname='Dent',
+                       email='adent@example.org'),
+                Author(order=1, forename='Ford', surname='Prefect',
+                       email='fprefect@example.org'),
+            ]),
+            SetLicense(creator=user, license_uri='http://creativecommons.org/publicdomain/zero/1.0/',
+                       license_name='Foo zero 1.0'),
+            SetPrimaryClassification(creator=user, category='astro-ph.GA'),
+            ConfirmPolicy(creator=user),
+            SetUploadPackage(creator=user, identifier='1'),
+            ConfirmContactInformation(creator=user),
+            FinalizeSubmission(creator=user)
+        ]
+        events2 = [
+            # second submission
+            CreateSubmission(creator=user),
+            SetTitle(creator=user, title='Bar title'),
+            SetAbstract(creator=user, abstract='Indubitably' * 20),
+            SetAuthors(creator=user, authors=[
+                Author(order=0, forename='Jane', surname='Doe',
+                       email='jadoe@example.com'),
+                Author(order=1, forename='John', surname='Doe',
+                       email='jodoe@example.com'),
+            ]),
+            SetLicense(creator=user, license_uri='http://creativecommons.org/publicdomain/zero/1.0/',
+                       license_name='Foo zero 1.0'),
+            SetPrimaryClassification(creator=user, category='astro-ph.GA'),
+            ConfirmPolicy(creator=user),
+            SetUploadPackage(creator=user, identifier='1'),
+            ConfirmContactInformation(creator=user),
+            FinalizeSubmission(creator=user)
+        ]
+
+        with in_memory_db():
+            # User creates and finalizes submission.
+            with transaction():
+                before = None
+                for i, event in enumerate(list(events1)):
+                    event.created = datetime.now(UTC)
+                    after = event.apply(before)
+                    event, after = store_event(event, before, after)
+                    events1[i] = event
+                    before = after
+                submission1 = after
+                ident1 = submission1.submission_id
+
+                before = None
+                for i, event in enumerate(list(events2)):
+                    event.created = datetime.now(UTC)
+                    after = event.apply(before)
+                    event, after = store_event(event, before, after)
+                    events2[i] = event
+                    before = after
+                submission2 = after
+                ident2 = submission2.submission_id
+
+            classic_sub = models.Submission(
+                type='new',
+                submitter_id=42)
+            session = current_session()
+            session.add(classic_sub)
```

Comment:
This may not be committed prior to ``submissions = get_user_submissions_fast(42)``. Can probably just move L227-231 into the preceding transaction block.